public/app/features/dashboard/dashgrid/PanelStateWrapper.tsx

```diff
@@ -547,6 +561,25 @@ export class PanelChrome extends PureComponent<Props, State> {
           }}
         </ErrorBoundary>
       </section>
+    ) : (
+      <PanelChrome width={width} height={height} title={title} leftItems={leftItems}>
+        {(innerWidth, innerHeight) => (
+          <>
+            <ErrorBoundary
+              dependencies={[data, plugin, panel.getOptions()]}
+              onError={this.onPanelError}
+              onRecover={this.onPanelErrorRecover}
+            >
+              {({ error }) => {
+                if (error) {
+                  return null;
+                }
+                return this.renderPanel(innerWidth, innerHeight);
```

Comment:
think you will need a different renderPanel here, PanelChrome already contains a lot of the logic (and div styles) that renderPanel does (size calculations that deducts size for heading, and takes into account if padding option etc), the panel-content wrappper div etc is all handled by PanelChrome

Reply:
Thanks, Torkel 🙌🏾  , I created a new [renderPanel function](https://github.com/grafana/grafana/pull/56659/commits/1b5a24bf57fd873da7dd34af0623eaa5f5fe5803#diff-343fa823fe1134aec1e55322686f607f10640288d4cca873d38b086c17924cffR441) which removed the div styles and all the stuff that `PanelChrome` takes care of.

I think I'll refactor this code tomorrow, maybe having a different component to render? (right now this component looks messy with the new changes). I was also playing around today using `PanelRenderer` but found it to be quite limited and some features were not working as expected (i.e. toggle legend on time series)

Reply:
small update 😄: after a sync with @dprokop, I decided to avoid refactoring this component(non-trivial), so this PR just made the changes of having a different function that renders the panel without the logic regarding headers, size calculations, etc.